Lib/test/test_socket.py

```diff
@@ -5970,6 +5970,21 @@ def test_sendmsg_afalg_args(self):
             with self.assertRaises(TypeError):
                 sock.sendmsg_afalg(op=socket.ALG_OP_ENCRYPT, assoclen=-1)
 
+    def test_length_restriction(self):
+        # bpo-35050, off-by-one error in length check
+        sock = socket.socket(socket.AF_ALG, socket.SOCK_SEQPACKET, 0)
+        # salg_type[14]
```

Comment:
I dislike this test since it seems too specific to one Linux version. But I guess that the size of these fields is not going to change. If they change, we can update the test.

Reply:
Yeah, it's part of the Kernel ABI and not likely to change any time.

Reply:
Ok, that's fair. And I guess that currently only Linux supports AF_ALG.